addon-chrome/ChromeActivityStreams.js

```diff
@@ -0,0 +1,243 @@
+const {CONTENT_TO_ADDON} = require("common/event-constants");
+const ChromePlacesProvider = require("addon-chrome/ChromePlacesProvider");
+const ChromeSearchProvider = require("addon-chrome/ChromeSearchProvider");
+const {dispatch} = require("addon-chrome/ChromeActionManager");
+
+module.exports = class ChromeActivityStreams {
```

Comment:
You're doing three types of things here.
1) subscribing to data changes from Chrome
2) routing "actions" from Activity Stream app
3) handling above actions

I would consider breaking these up into separate modules. You will likely end up with two or three modules, depending on if you want to combine routing and app handling of actions.